### ISSUE TITLE
Alert check skipped, NPE

### ISSUE THREAD

**Issue by hc4** [Graylog2/graylog2-server#4045]
Just upgraded to 2.3.0 and got lots of errors in log
```
2017-08-02T10:54:55.727+03:00 ERROR [AlertScanner] Skipping alert check <***/988cee81-212a-44c3-9477-289e64e530cf>: null (NullPointerException)
2017-08-02T10:55:54.576+03:00 ERROR [AlertScanner] Skipping alert check <***/988cee81-212a-44c3-9477-289e64e530cf>: null (NullPointerException)
2017-08-02T10:56:56.852+03:00 ERROR [AlertScanner] Skipping alert check <***/03cd63aa-94c8-4d1c-8e47-c02bed2f2e7d>: null (NullPointerException)
2017-08-02T10:57:02.803+03:00 ERROR [AlertScanner] Skipping alert check <***/988cee81-212a-44c3-9477-289e64e530cf>: null (NullPointerException)
2017-08-02T10:57:55.756+03:00 ERROR [AlertScanner] Skipping alert check <***/03cd63aa-94c8-4d1c-8e47-c02bed2f2e7d>: null (NullPointerException)
2017-08-02T10:57:55.881+03:00 ERROR [AlertScanner] Skipping alert check <***/988cee81-212a-44c3-9477-289e64e530cf>: null (NullPointerException)
```

## Your Environment

* Graylog Version: 2.3.0
* Elasticsearch Version: 2.4.5

**Comment by joschi** [comment]
@hc4 Are there any other exceptions in the logs of your Graylog node(s)?
Are you using 3rd party plugins and are all plugins compatible with Graylog 2.3.0?

**Comment by hc4** [comment]
No other errors in logs (also surprised that there  is no stacktrace).
![image](https://user-images.githubusercontent.com/17475270/28863802-0459afba-7773-11e7-9cb6-6d3c7c44bb59.png)

**Comment by hc4** [comment]
It looks like problem caused by broken stream routing (another problem, that I'm looking right now)
So there is no any data in alert stream.

**Comment by dennisoelkers** [comment]
Hey @hc4, can you dump your alert condition configurations?

**Comment by hc4** [comment]
```
{
  "total": 2,
  "conditions": [
    {
      "id": "f0192668-c8c8-4eb6-a765-f66a1919431c",
      "type": "message_count",
      "creator_user_id": "***",
      "created_at": "2016-09-07T10:32:25.889+0000",
      "parameters": {
        "grace": 1440,
        "threshold_type": "LESS",
        "threshold": 1,
        "time": 60,
        "backlog": 0
      },
      "in_grace": false,
      "title": "***"
    },
    {
      "id": "988cee81-212a-44c3-9477-289e64e530cf",
      "type": "field_value",
      "creator_user_id": "***",
      "created_at": "2016-10-17T08:31:20.739+0000",
      "parameters": {
        "backlog": 0,
        "repeat_notifications": false,
        "field": "load_percents",
        "grace": 60,
        "threshold": 90,
        "threshold_type": "HIGHER",
        "time": 5,
        "type": "MEAN"
      },
      "in_grace": false,
      "title": "***"
    }
  ]
}
```

**Comment by dennisoelkers** [comment]
Can you please: 

- also dump the config for alert condition 03cd63aa-94c8-4d1c-8e47-c02bed2f2e7d
- switch the AlertScanner class to debug for one run, using `curl -XPUT -u $user:$pass http://$graylog/api/system/loggers/org.graylog2.alerts.AlertScanner/DEBUG` (replace http with https if necessary)

When switched to debug, the whole exception is logged. Otherwise it is only the exception message and class name, but no stack trace, to avoid log flooding. You can switch back the log level of the class by doing the same call, replacing `DEBUG` with `INFO`.

**Comment by hc4** [comment]
```
java.lang.NullPointerException: null
        at org.graylog2.indexer.results.FieldStatsResult.<init>(FieldStatsResult.java:50) ~[graylog.jar:?]
        at org.graylog2.indexer.searches.Searches.fieldStats(Searches.java:508) ~[graylog.jar:?]
        at org.graylog2.alerts.types.FieldValueAlertCondition.runCheck(FieldValueAlertCondition.java:175) ~[graylog.jar:?]
        at org.graylog2.alerts.types.FieldValueAlertCondition.runCheck(FieldValueAlertCondition.java:53) ~[graylog.jar:?]
        at org.graylog2.alerts.AlertScanner.checkAlertCondition(AlertScanner.java:64) ~[graylog.jar:?]
        at org.graylog2.periodical.AlertScannerThread.lambda$doRun$0(AlertScannerThread.java:63) ~[graylog.jar:?]
        at java.util.ArrayList.forEach(ArrayList.java:1249) [?:1.8.0_131]
        at org.graylog2.periodical.AlertScannerThread.doRun(AlertScannerThread.java:63) [graylog.jar:?]
        at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:77) [graylog.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_131]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_131]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_131]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
```

**Comment by dennisoelkers** [comment]
Thanks. This is a known issue (https://github.com/Graylog2/graylog2-server/pull/4038) and will be fixed in the next version.

**Comment by dennisoelkers** [comment]
@hc4: I have backported the fix to `2.3`, so the fix should go into `2.3.1` if everything goes well. Thanks for reporting this!

**Comment by joschi** [comment]
Fixed in #4046
